neohtop 1.0.8 (new cask)

junminhong opened the pull request:
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

junminhong commented:
Oh no, this repository is too young

khipp commented on Casks/n/neohtop.rb:
```diff
@@ -0,0 +1,29 @@
+cask "neohtop" do
+  version "1.0.5"
+
+  on_arm do
+    sha256 "eea9bf2a112dc0cf978a7d1ae87608d98d5eccc248f1057469959cfd630f0b46"
+
+    url "https://github.com/Abdenasser/neohtop/releases/download/v#{version}/NeoHtop-silicon.dmg", verified: "github.com/Abdenasser/neohtop/"
+  end
+  on_intel do
+    sha256 "f29b038e64f9c73c3f756004f2d798b207b109e3c270f04f162669247ccf49bb"
+
+    url "https://github.com/Abdenasser/neohtop/releases/download/v#{version}/NeoHtop-intel.dmg", verified: "github.com/Abdenasser/neohtop/"
+  end
+
```
```suggestion
  arch arm: "silicon", intel: "intel"

  version "1.0.5"
  sha256 arm:   "eea9bf2a112dc0cf978a7d1ae87608d98d5eccc248f1057469959cfd630f0b46",
         intel: "f29b038e64f9c73c3f756004f2d798b207b109e3c270f04f162669247ccf49bb"

  url "https://github.com/Abdenasser/neohtop/releases/download/v#{version}/NeoHtop-#{arch}.dmg",
      verified: "github.com/Abdenasser/neohtop/"
```

junminhong replied:
Thanks for the review, fix 🚧 [cee3148](https://github.com/Homebrew/homebrew-cask/pull/191295/commits/cee3148feed059d82589df8744634a6d9f88047c)

khipp commented on Casks/n/neohtop.rb:
```diff
@@ -0,0 +1,29 @@
+  name "NeoHtop"
+  desc "Htop on steroids"
+  homepage "https://abdenasser.github.io/neohtop/"
+
+  livecheck do
+    url :url
+    strategy :github_latest
+  end
+
```
```suggestion

```

junminhong replied:
Thanks for the review! I've implemented the fixes as suggested. 🚧 Commit [cee3148](https://github.com/Homebrew/homebrew-cask/pull/191295/commits/cee3148feed059d82589df8744634a6d9f88047c). Let me know if there’s anything else that needs adjustment.

khipp commented on Casks/n/neohtop.rb:
```diff
@@ -0,0 +1,29 @@
+    strategy :github_latest
+  end
+
+  depends_on macos: ">= :catalina"
```
```suggestion
  depends_on macos: ">= :high_sierra"
```

junminhong replied:
Thanks for the review, fix 🚧 [cee3148](https://github.com/Homebrew/homebrew-cask/pull/191295/commits/cee3148feed059d82589df8744634a6d9f88047c)

khipp reviewed (commented):
Thank you for your first contribution to `homebrew-cask`, @junminhong! 🎉

I left a few suggestions below.

khipp commented on Casks/n/neohtop.rb:
```diff
@@ -0,0 +1,29 @@
+
+  app "NeoHtop.app"
+
+  zap trash: "~/Library/Caches/com.neohtop.dev"
```
```suggestion
  zap trash: [
    "~/Library/Caches/com.neohtop.dev",
    "~/Library/Saved Application State/com.neohtop.dev.savedState",
    "~/Library/WebKit/com.neohtop.dev",
  ]
```

daeho-ro commented:
Github latest release is marked on the nightly build and so I request it to change.
- https://github.com/Abdenasser/neohtop/issues/46

daeho-ro commented:
@junminhong I am so sorry for this but the upstream changes their pipeline and so we have to change the artifact url to the tag artifacts. See here,
- https://github.com/Abdenasser/neohtop/releases/tag/v1.0.8

Since the artifact has two arch dependent prefix and suffix, another variable is required. Can you update it?

Abdenasser commented:
@daeho-ro @junminhong you might need to slightly change the URLs, this is the format we'll use from now on.
[https://github.com/Abdenasser/neohtop/releases/tag/v1.0.8](https://github.com/Abdenasser/neohtop/releases/tag/v1.0.8)

junminhong commented:
> @junminhong I am so sorry for this but the upstream changes their pipeline and so we have to change the artifact url to the tag artifacts. See here,
> 
> * https://github.com/Abdenasser/neohtop/releases/tag/v1.0.8
> 
> Since the artifact has two arch dependent prefix and suffix, another variable is required. Can you update it?

Thank you for the heads-up! I’ve updated the artifact URL to use the tag-based structure, and added the additional variable to handle the arch-dependent prefix and suffix. Here’s the commit with the fix: 🚧 [273af08](https://github.com/Homebrew/homebrew-cask/pull/191295/commits/273af0888afe8af41a3467b9db3289ae584607e0).

junminhong commented:
I'm not sure whether I should use https://github.com/Abdenasser/neohtop/releases/download/ or https://github.com/Abdenasser/neohtop/releases/tag/ as the URL.

And, do I need to add this livecheck code back in?
<img width="566" alt="Screenshot 2024-11-08 at 10 16 07" src="https://github.com/user-attachments/assets/094b3cca-0eff-461d-a8bb-c6c1c9954f81">
```
livecheck do
  url :url
  strategy :github_latest
end
```

daeho-ro commented:
@junminhong I left a comment on the code, we have to add livecheck again for filter the tag values.

junminhong commented:
> @junminhong I left a comment on the code, we have to add livecheck again for filter the tag values.

🚧 [00c65c3](https://github.com/Homebrew/homebrew-cask/pull/191295/commits/00c65c3f5fe52148daf85b8a69ed8d7d0c7d951e)

daeho-ro commented:
As a final, I squashed your commits. Thank you @junminhong !
